src/app/pages/PerpetualPage/hooks/usePerpetual_openTrade.ts

```diff
@@ -1,28 +1,57 @@
 import { useAccount } from 'app/hooks/useAccount';
 import { useSendContractTx } from 'app/hooks/useSendContractTx';
-import { BigNumber } from 'ethers';
-import { bignumber } from 'mathjs';
+import { useMemo } from 'react';
 import { TxType } from 'store/global/transactions-store/types';
+import { TradingPosition } from 'types/trading-position';
+import { fromWei } from 'utils/blockchain/math-helpers';
 import { ethGenesisAddress, gasLimit } from 'utils/classifiers';
-import { floatToABK64x64, PERPETUAL_ID } from '../utils';
+import {
+  calculateSlippagePrice,
+  getIndexPrice,
+  getRequiredMarginCollateral,
+} from '../temporaryUtils';
+import { floatToABK64x64, getTradeDirection, PERPETUAL_ID } from '../utils';
 import { usePerpetual_depositMarginToken } from './usePerpetual_depositMarginToken';
+import { usePerpetual_marginAccountBalance } from './usePerpetual_marginAccountBalance';
+import { usePerpetual_queryAmmState } from './usePerpetual_queryAmmState';
+import { usePerpetual_queryPerpParameters } from './usePerpetual_queryPerpParameters';
 
-const MASK_MARKET_ORDER = BigNumber.from('0x40000000');
+const MASK_MARKET_ORDER = 0x40000000;
 const MASK_CLOSE_ONLY = 0x80000000;
 
 export const usePerpetual_openTrade = () => {
   const address = useAccount();
+  const perpetualParameters = usePerpetual_queryPerpParameters();
+  const ammState = usePerpetual_queryAmmState();
+  const marginBalance = usePerpetual_marginAccountBalance();
+  const indexPrice = useMemo(() => getIndexPrice(ammState), [ammState]);
 
   const { deposit } = usePerpetual_depositMarginToken();
   const { send, ...rest } = useSendContractTx('perpetualManager', 'trade');
 
   return {
-    trade: async (marginAmount: string, totalAmount: string) => {
-      const isClosePositionTransaction = bignumber(totalAmount).lt(0);
-      const limitPrice = isClosePositionTransaction ? -100000 : 1000000; // this is hardcoded for now but Vasili is preparing a function for this
+    trade: async (
+      isClosePosition: boolean | undefined = false,
+      amount: number | undefined = 0,
+      leverage: number | undefined = 1,
+      slippage: number | undefined = 0.5,
+      tradingPosition: TradingPosition | undefined = TradingPosition.LONG,
+    ) => {
+      const limitPrice = calculateSlippagePrice(
+        indexPrice,
+        slippage,
+        getTradeDirection(tradingPosition),
+      );
+
+      const marginCollateralAmount = getRequiredMarginCollateral(
+        leverage,
+        marginBalance.fPositionBC,
+        amount,
+        perpetualParameters,
+      );
 
-      if (!isClosePositionTransaction) {
-        await deposit(marginAmount);
+      if (!isClosePosition) {
+        await deposit(String(marginCollateralAmount));
```

Comment:
Is marginCollateralAmount a wei amount? Otherwise you need to use toWei as deposit expects a wei amount, doesn't it?

Reply:
`marginCollateralAmount` is already wei amount (but it's a number and not a string).